pom.xml

```diff
@@ -285,6 +285,97 @@
 
     <build>
         <plugins>
+            <plugin>
+                <groupId>biz.aQute.bnd</groupId>
+                <artifactId>bnd-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>bnd-process</id>
+                        <configuration>
+                            <bnd><![CDATA[
+Bundle-Name: ${project.name}
+Bundle-SymbolicName: ${project.artifactId}
+Bundle-DocURL: http://sling.apache.org/site/logging.html
+Import-Package: org.osgi.service.cm;version="[1.2,2)";resolution:=dynamic, \
+    org.osgi.service.event;version="[1.2,2)";resolution:=dynamic, \
+    javax.xml.transform.*;resolution:=dynamic, \
+    org.slf4j.bridge;resolution:=optional, \
+    *
+
+Require-Capability: osgi.extender;filter:="(&(osgi.extender=osgi.serviceloader.registrar)(version>=1.0.0)(!(version>=2.0.0)))"
+Provide-Capability: osgi.serviceloader;osgi.serviceloader="ch.qos.logback.classic.spi.Configurator"
+]]></bnd>
+                            <manifestPath>${project.build.directory}/nodeps-classes/META-INF/MANIFEST.MF</manifestPath>
+                            <outputDir>${project.build.directory}/nodeps-classes</outputDir>
+                        </configuration>
+                    </execution>
+                    <execution>
+                        <id>bnd-process-deps</id>
+                        <goals>
+                            <goal>bnd-process</goal>
+                        </goals>
+                        <configuration>
+                            <bnd><![CDATA[
+Bundle-Name: ${project.name} (Including dependencies)
+Bundle-SymbolicName: ${project.artifactId}.all
+Bundle-DocURL: http://sling.apache.org/site/logging.html
+-includeresource:@logback-core-[0-9.]*(-SNAPSHOT)?.jar, \
+    @logback-classic-[0-9.]*(-SNAPSHOT)?.jar, \
+    @slf4j-api-[0-9.]*(-SNAPSHOT)?.jar
+Import-Package: org.osgi.service.cm;version="[1.2,2)";resolution:=dynamic, \
+    org.osgi.service.event;version="[1.2,2)";resolution:=dynamic, \
+    javax.xml.transform.*;resolution:=dynamic, \
+    org.slf4j.bridge;resolution:=optional, \
+    jakarta.mail;resolution:=optional, \
+    jakarta.mail.internet;resolution:=optional, \
+    jakarta.servlet;resolution:=optional, \
+    jakarta.servlet.http;resolution:=optional, \
+    org.codehaus.commons.compiler;resolution:=optional;version="[3.1,4)", \
+    org.codehaus.janino;resolution:=optional;version="[3.1,4)", \
+    org.fusesource.jansi;resolution:=optional;version="[2.4,3)", \
+    org.apache.commons.lang3;resolution:=optional;version="[3.0,4)", \
```

Comment:
Is it necessary to have explicit version numbers in the imports? I am concerned that this will be difficult and error prone to maintain. If bnd could determine these versions automatically, that would be preferable.

Reply:
It does not compute them, the problem is that the dependency is missing and bnd has no knowledge about that version. Now adding the dependency to the pom would solve that issue, but then you need to keep the dependency in sync - which is the same.
Agree it is error prone, the only option I see right now is to add a big comment above